include/oneapi/dpl/experimental/kt/two_pass_scan.h

```diff
@@ -174,48 +228,81 @@ two_pass_scan(sycl::queue q, _InRng&& __in_rng, _OutRng&& __out_rng,
                 auto sub_group_id = sub_group.get_group_linear_id();
                 auto sub_group_local_id = sub_group.get_local_linear_id();
 
-                ValueType v = identity;
-                ValueType sub_group_carry = identity;
+                oneapi::dpl::__par_backend_hetero::__lazy_ctor_storage<ValueType> sub_group_carry;
+                std::size_t group_start_idx = (b * blockSize) + (g * K * num_sub_groups_local);
+                if (M <= group_start_idx)
+                    return; // exit early for empty groups (TODO: avoid launching these?)
 
-                size_t start_idx = (b * blockSize) + (g * K * num_sub_groups_local) + (sub_group_id * K);
-                bool is_full_thread = start_idx + J * VL <= M;
-                // adjust for lane-id
-                start_idx += sub_group_local_id;
-                // compute sub-group local pfix on T0..63, K samples/T, send to accumulator kernel
-                if (is_full_thread && is_full_block)
+                    
+                std::size_t elements_in_group = std::min(M - group_start_idx, std::size_t(num_sub_groups_local * K));
+                auto active_subgroups = oneapi::dpl::__internal::__dpl_ceiling_div(elements_in_group, K);
+                std::size_t subgroup_start_idx = group_start_idx + (sub_group_id * K);
+                bool is_full_thread = subgroup_start_idx + J * VL <= M;
+                size_t start_idx = subgroup_start_idx + sub_group_local_id;
+
+                if (sub_group_id < active_subgroups)
                 {
-                    _ONEDPL_PRAGMA_UNROLL
-                    for (int j = 0; j < J_max; j++)
+                    // adjust for lane-id
+                    // compute sub-group local pfix on T0..63, K samples/T, send to accumulator kernel
+                    if (is_full_thread && is_full_block)
                     {
-                        v = unary_op(__in_rng[start_idx + j * VL]);
+                        auto v = unary_op(__in_rng[start_idx]);
                         // In principle we could use SYCL group scan. Stick to our own for now for full control of implementation.
-                        std::tie(v, sub_group_carry) = sub_group_scan<VL, Inclusive>(sub_group, std::move(v), binary_op, std::move(sub_group_carry));
+                        sub_group_scan<VL, Inclusive, false>(sub_group, v, binary_op, sub_group_carry);
```

Comment:
For calls to `sub_group_scan`, can we annotate the last template parameter. For example:
```suggestion
                        sub_group_scan<VL, Inclusive, /*InitPresent=*/false>(sub_group, v, binary_op, sub_group_carry);
```

Reply:
good suggestion.  Added labels for unclear template params to `sub_group_scan*`